cli/src/utils/subprocess.ts

```diff
@@ -0,0 +1,44 @@
+import { ChildProcess, spawn as origSpawn, SpawnOptions } from 'child_process';
+
+export class ChildProcessError extends Error {
+  public readonly name = 'ChildProcessError';
+  public command?: string;
+  public args?: ReadonlyArray<string>;
+
+  constructor(
+    message: string,
+    public readonly code: number | null,
+    public readonly signal: string | null,
+  ) {
+    super(message);
+  }
+}
+
+export async function spawn(
```

Comment:
Might be simpler to use `execFile(Sync)` instead of a wrapper?

Reply:
I like that you can await this version.